[2024-02-26T09:01:47.389+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: exchange_trade_data.connect_to_md manual__2024-02-26T09:01:43.010468+00:00 [queued]>
[2024-02-26T09:01:47.414+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: exchange_trade_data.connect_to_md manual__2024-02-26T09:01:43.010468+00:00 [queued]>
[2024-02-26T09:01:47.417+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2024-02-26T09:01:47.824+0000] {taskinstance.py:2191} INFO - Executing <Task(_PythonDecoratedOperator): connect_to_md> on 2024-02-26 09:01:43.010468+00:00
[2024-02-26T09:01:47.845+0000] {standard_task_runner.py:60} INFO - Started process 8710 to run task
[2024-02-26T09:01:47.847+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'exchange_trade_data', 'connect_to_md', 'manual__2024-02-26T09:01:43.010468+00:00', '--job-id', '110', '--raw', '--subdir', 'DAGS_FOLDER/exchange.py', '--cfg-path', '/tmp/tmpx_sy_zk7']
[2024-02-26T09:01:47.849+0000] {standard_task_runner.py:88} INFO - Job 110: Subtask connect_to_md
[2024-02-26T09:01:48.003+0000] {task_command.py:423} INFO - Running <TaskInstance: exchange_trade_data.connect_to_md manual__2024-02-26T09:01:43.010468+00:00 [running]> on host f5283f9a0a92
[2024-02-26T09:01:48.179+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='exchange_trade_data' AIRFLOW_CTX_TASK_ID='connect_to_md' AIRFLOW_CTX_EXECUTION_DATE='2024-02-26T09:01:43.010468+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-02-26T09:01:43.010468+00:00'
[2024-02-26T09:01:52.481+0000] {python.py:201} INFO - Done. Returned value was: <duckdb.duckdb.DuckDBPyConnection object at 0x7ff54236ee30>
[2024-02-26T09:01:52.490+0000] {xcom.py:664} ERROR - Object of type DuckDBPyConnection is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config or make sure to decorate your object with attr.
[2024-02-26T09:01:52.493+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/json.py", line 91, in default
    return serialize(o)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/serialization/serde.py", line 180, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'duckdb.duckdb.DuckDBPyConnection'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 440, in _execute_task
    task_instance.xcom_push(key=XCOM_RETURN_KEY, value=xcom_value, session=session)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2980, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/xcom.py", line 247, in set
    value = cls.serialize_value(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/xcom.py", line 662, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
  File "/usr/local/lib/python3.8/json/__init__.py", line 234, in dumps
    return cls(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
  File "/usr/local/lib/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/local/lib/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/json.py", line 93, in default
    return super().default(o)
  File "/usr/local/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type DuckDBPyConnection is not JSON serializable
[2024-02-26T09:01:52.506+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=exchange_trade_data, task_id=connect_to_md, execution_date=20240226T090143, start_date=20240226T090147, end_date=20240226T090152
[2024-02-26T09:01:52.526+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 110 for task connect_to_md (Object of type DuckDBPyConnection is not JSON serializable; 8710)
[2024-02-26T09:01:52.540+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-02-26T09:01:52.583+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
